tests/smoke-tests/test-staking-rewards.ts

```diff
@@ -6,23 +6,307 @@ import { ApiPromise } from "@polkadot/api";
+    const estimatedTime = (promises.length / 300).toFixed(2);
+    debug(
+      "Verifying staked amounts for " +
+        promises.length +
+        " delegates, estimated time " +
+        estimatedTime +
+        " mins."
+    );
+
+    const results = await Promise.all(promises);
+    const mismatches = results.filter((item) => item.match == false);
+    expect(
+      mismatches,
+      `Mismatched amounts for ${mismatches
+        .map((a) => `delegator ${a.delegator} collator:${a.collator}`)
+        .join(", ")}`
+    ).to.be.empty;
+
+    // Exit buffer for cleanup
+    await new Promise((resolve) => setTimeout(resolve, 2000));
+  });
+
+  it("should snapshot delegate autocompound preferences correctly.", async function () {
+    // This test is so slow due to rate limiting, this should be off until a better solution appears
+    if (process.env.RUN_ATSTAKE_CONSISTENCY_TESTS != "true") {
+      debug("Explicit RUN_ATSTAKE_CONSISTENCY_TESTS flag not set to 'true', skipping test");
+      this.skip();
+    }
+    const specVersion = context.polkadotApi.consts.system.version.specVersion.toNumber();
+    if (specVersion < 1900) {
+      debug(`Autocompounding not supported for ${specVersion}, skipping test.`);
+      this.skip();
+    }
+
+    this.timeout(ONE_HOURS);
+    this.slow(ONE_HOURS);
+    const limiter = new Bottleneck({
+      maxConcurrent: 10,
+      minTime: 100,
     });
+
+    // Function to check a single Delegator's delegation to a collator
+    const checkDelegatorAutocompound = async (collatorId: AccountId20, delegatorSnapshot) => {
+      const autoCompoundQuery = (
+        (await limiter.schedule(() =>
+          predecessorApiAt.query.parachainStaking.autoCompoundingDelegations(collatorId)
+        )) as any
+      ).find((a) => a.delegator.toString() == delegatorSnapshot.owner.toString());
+      const autoCompoundAmount =
+        autoCompoundQuery == undefined ? new BN(0) : autoCompoundQuery.value;
+      const match = autoCompoundAmount.eq(delegatorSnapshot.autoCompound);
+      if (!match) {
+        debug(
+          "Snapshot autocompound " +
+            delegatorSnapshot.autoCompound.toString() +
+            "% does not match storage autocompound " +
+            autoCompoundAmount.toString() +
+            "% for delegator: " +
+            delegatorSnapshot.owner.toString() +
+            " on candidate: " +
+            collatorId.toString()
+        );
+      }
+      return {
+        collator: collatorId.toString(),
+        delegator: delegatorSnapshot.owner.toString(),
+        match,
+      };
+    };
+
+    debug(`Gathering snapshot query requests for ${atStakeSnapshot.length} collators.`);
+    const promises = atStakeSnapshot
+      .map((coll) => {
+        const [
+          {
+            args: [_, accountId],
+          },
+          { bond, total, delegations },
+        ] = coll;
+        return delegations.map((delegation) => checkDelegatorAutocompound(accountId, delegation));
```

Comment:
We'd probably perform too many storage reads here reading each delegator for each collator. This can be optimized by fetching the `predecessorApiAt.query.parachainStaking.autoCompoundingDelegations(collatorId)` only once and caching it (and/or create a lookup)

Reply:
Great idea, it's a tonne faster now.

Reply:
Great idea, it's a tonne faster now.